redwood-tooling-codegen/src/main/kotlin/app/cash/redwood/tooling/codegen/testingGeneration.kt

```diff
@@ -281,6 +287,23 @@ internal fun generateWidgetValue(schema: Schema, widget: Widget): FileSpec {
     .add("layoutModifiers,\n")
   val toStringBuilder = StringBuilder()
     .append("${widgetValueType.simpleName}(layoutModifiers=${'$'}layoutModifiers")
+  val childrenDiff = MemberName("app.cash.redwood.protocol", "ChildrenDiff")
+  val addToBuilder = CodeBlock.builder()
+    .addStatement("val widgetId = Id(builder.nextId++)")
+    .addStatement(
+      "val widgetTag: WidgetTag = %L",
+      buildCodeBlock {
+        val wt = MemberName("app.cash.redwood.protocol", "WidgetTag")
+        if (widget is ProtocolWidget) {
+          add("%M(${widget.tag})", wt)
+        }
+      },
+    )
```

Comment:
```suggestion
    .addStatement("val widgetTag = %T(%L)", Protocol.WidgetTag, (widget as ProtocolWidget).tag)
```
KotlinPoet will throw an exception if the existing `if` block is not taken so switching to an `as` cast will throw earlier in the process and with an exception that indicates what the problem is.

For now, there's _only_ going to be `ProtocolWidget` instances as you can't do a non-protocol schema yet.